src/components/appointments/CreateAppointment/CreateAppointmentState.js

```diff
@@ -8,25 +9,75 @@ import {CreateAppointmentContext} from '../Context';
 
 const SESSION_STORAGE_KEY = 'appointment|formData';
 
-export const CreateAppointmentState = ({currentStep, submission, children}) => {
-  const [appointmentData, setAppointmentData] = useSessionStorage(SESSION_STORAGE_KEY, {});
+const errorKeysByStep = {
+  producten: ['products'],
+  kalender: ['location', 'date', 'datetime'],
+  contactgegevens: ['contactDetails'],
+};
+
+export const buildContextValue = (
+  submission,
+  currentStep,
+  appointmentData,
+  setAppointmentData = () => {},
+  appointmentErrors = {initialTouched: {}, initialErrors: {}},
+  setAppointmentErrors = () => {}
+) => {
   const submittedSteps = Object.keys(appointmentData).filter(
     subObject => Object.keys(subObject).length
   );
   const mergedAppointmentData = Object.keys(appointmentData).reduce(
     (accumulator, key) => ({...accumulator, ...appointmentData[key]}),
     {}
   );
+
+  const errorKeys = errorKeysByStep[currentStep];
+  const {initialTouched, initialErrors} = appointmentErrors;
+
+  const stepInitialTouched = {};
+  const stepInitialErrors = {};
+  errorKeys.forEach(key => {
+    const errors = initialErrors[key];
+    if (!errors) return;
```

Comment:
it can be a multitude of things - `undefined`, a nested object or a nested list. However, the exact datastructure is not relevant, as only the prefix of the error matters.

E.g. the step choose product only cares about the `products` key, which is a list of products (ID + amount). But, if there any validation errors, they will always be present as `{"products": [undefined, {"amount": "'..."}]}` or be absent if there are no errors. 

For the second step, the fields/prefixes `location`, `date` and `datetime` are relevant and those happen to just be mappings of error -> string. Finally, `contactDetails` the container for the third step.

This utility ensures that a complete error structure like:

```js
{
  "products": [{"productId": "nope"}],
  "location": "cantdothat",
  "contactDetails": {
    "dateOfBirth": "people over > 200 years old don't exist",
  },
}
```
for step 1 only includes the `products` errors, for step 2 only the `location` error and for step 3 only the `contactDetails` errors.